prow/cmd/deck/main.go

```diff
@@ -195,29 +211,40 @@ func localOnlyMain(cfg config.Getter, o options, mux *http.ServeMux) *http.Serve
 	return mux
 }
 
+type podLogClient struct {
+	client corev1.PodInterface
+}
+
+func (c *podLogClient) GetLogs(name string, opts *coreapi.PodLogOptions) ([]byte, error) {
+	reader, err := c.client.GetLogs(name, &coreapi.PodLogOptions{Container: kube.TestContainerName}).Stream()
+	if err != nil {
+		return nil, err
+	}
+	defer reader.Close()
+	return ioutil.ReadAll(reader)
+}
+
 // prodOnlyMain contains logic only used when running deployed, not locally
 func prodOnlyMain(cfg config.Getter, o options, mux *http.ServeMux) *http.ServeMux {
-	kc, err := kube.NewClientInCluster(cfg().ProwJobNamespace)
+	kc, defaultContext, kubernetesClients, err := o.kubernetes.Client(cfg().ProwJobNamespace, false)
```

Comment:
Why `dryrun == false`? Since deck doesn't make modifying actions can we set this to `true`?

Reply:
We should leave it false for consistency. If we don't make any modifying actions it shouldn't matter, right?

Reply:
Yeah, the idea was what Erick said